libs/features/components/ui-lists/src/lib/components/elewa-group-horizontal-timeline-carousel/elewa-group-horizontal-timeline-carousel.component.ts

```diff
@@ -0,0 +1,208 @@
+  }
+}
+
+// CHECK IF AN ELEMENT IS IN VIEWPORT
+public isElementInViewport(el: HTMLElement, position: string) {
+  const rect = el.getBoundingClientRect();
+  if(position == 'first'){
+    return (
+      rect.top >= 0 &&
+      rect.left + 50 >= 0 &&
+      rect.bottom <=
+        (window.innerHeight || document.documentElement.clientHeight) &&
+      rect.right <= (window.innerWidth || document.documentElement.clientWidth)
+    );
+  }
+  return (
+    rect.top >= 0 &&
+    rect.left + 50 >= 0 &&
+    rect.bottom <=
+      (window.innerHeight || document.documentElement.clientHeight) &&
+    rect.right <= (window.innerWidth || document.documentElement.clientWidth)
+  );
+}
+
+// SET STATE OF PREV/NEXT ARROWS
+public setBtnState(el: HTMLButtonElement, flag = true) {
+  if (flag) {
+    el.classList.add(this.disabledClass);
+  } else {
+    if (el.classList.contains(this.disabledClass)) {
+      el.classList.remove(this.disabledClass);
+    }
+    el.disabled = false;
+  }
+}
+
+// ANIMATE TIMELINE
+public animateTl(
+  scrolling: number,
+  el: NodeListOf<HTMLButtonElement>,
+  tl: HTMLOListElement
+) {
+  let counter = 0;
+  console.log(`Line: 144`, el);
+  for (let i = 0; i < el.length; i++) {
+    el[i].addEventListener('click', () => {
+      if (!this.arrowPrev.disabled) {
+        this.arrowPrev.disabled = true;
+      }
+      if (!this.arrowNext?.disabled) {
+        this.arrowNext.disabled = true;
+      }
+      const sign = el[i].classList.contains('arrow__prev') ? '' : '-';
+      console.log(`Line: 155`, sign);
+      if (counter === 0) {
+        tl.style.transform = `translateX(-${scrolling}px)`;
+      } else {
+        const tlStyle = getComputedStyle(tl);
+        // add more browser prefixes if needed here
+        const tlTransform =
+          tlStyle.getPropertyValue('-webkit-transform') ||
+          tlStyle.getPropertyValue('transform');
+        const values =
+          parseInt(tlTransform.split(',')[4]) +
+          parseInt(`${sign}${scrolling}`);
+        tl.style.transform = `translateX(${values}px)`;
+      }
+      setTimeout(() => {
+        this.isElementInViewport(this.firstItem, 'first')
+          ? this.setBtnState(this.arrowPrev)
+          : this.setBtnState(this.arrowPrev, false);
+
+        this.isElementInViewport(this.lastItem, 'last')
+          ? this.setBtnState(this.arrowNext)
+          : this.setBtnState(this.arrowNext, false);
+      }, 100);
+      counter++;
+    });
+  }
+}
+
+// ADD BASIC KEYBOARD FUNCTIONALITY
+public setKeyboardFn(prev: HTMLButtonElement, next: HTMLButtonElement) {
+  document.addEventListener('keydown', (e) => {
+    if (e.which === 37 || e.which === 39) {
+      const timelineOfTop = this.timeline.offsetTop;
+      const y = window.pageYOffset;
+      if (timelineOfTop !== y) {
+        window.scrollTo(0, timelineOfTop);
+      }
+      if (e.which === 37) {
+        prev.click();
+      } else if (e.which === 39) {
+        next.click();
+      }
+    }
+  });
+}
+}
```

Comment:
we keep our code <=120 lines. These is a lot of code for just a carousel. There is a much better way of doing this with relatively less code.